core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverJobFactory.java

```diff
@@ -0,0 +1,37 @@
+package ai.timefold.solver.core.api.solver;
+
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+/**
+ * Provides a fluid contract that allows customization and submission of planning problems to solve.
+ * <p>
+ * To create a {@link SolverJob}, use {@link #solve()}.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}.
+ */
+public interface SolverJobFactory<Solution_, ProblemId_> {
```

Comment:
I understand why you decided to make this two interfaces, since you want to - correctly - share some of these methods.

In this case, there is another factor to consider - in the public API, we want to have interfaces which are easy to read and understand. If an interface requires another interface, that ease of reading goes down. I'd prefer if we avoided inheritance in this case.

Reply:
Let's wait for the design meeting to define the next steps here.

Reply:
Done!